src/lib/rest/restServiceLookup.cpp

```diff
@@ -97,5 +92,28 @@ RestService* restServiceLookup(ConnectionInfo* ciP, bool* badVerbP)
     ++serviceIx;
   }
 
-  return &serviceV[serviceIx];
+  if (match == true)
+  {
+    if (serviceV == restBadVerbV)
+    {
+      ciP->httpStatusCode = SccBadVerb;
+      ciP->badVerb        = true;
+    }
+
+    return &serviceV[serviceIx];
+  }
+
+  //
+  // Not found?
+  // Recursive call with bad verb service vector
+  // But only one recursive call !
+  //
+
+  if (*badVerbP == false)
+  {
+    *badVerbP = true;
+    return restServiceLookup(ciP, badVerbP);
+  }
+
+  return &restBadVerbV[104];  // FIXME 02: 99.9% sure we never get here, but if ... Change the 104 service for a fixed one
```

Comment:
From where the magic number 104 did come?

Reply:
yeah, this is beyond ugly.
I am sure we never get there, as all possible errors are caught before.
That said, `&restBadVerbV[104]` is a pointer to the last restservice item in the badVerb vector, the one that takes care of any error, not yet found. 

As the fixme says, should be changed for a *fixed* struct for this case.

fixme in place, not really in a hurry to implement it

Reply:
Also, once the new way of getting the info for "bad verb" (step 3 in issue #3109) this will go away